File: src/app/Forest/model/OffreRestaurant.ts
import { AccessRestaurant } from './accessRestaurant';
import { Restaurant } from './Restaurant';

export class OffreRestaurant {
  idOffreRestaurant: number;
  nameOffre: string;
  nbrDays: number;
  breakfast: boolean;
  lunch: boolean;
  dinner: boolean;
  price: number;
  accessRestaurants: AccessRestaurant[];
  restaurant: Restaurant;
}

File: Forest-Back/FoRest - Backend/src/main/java/esprit/DevUp/FoRest/Entity/OffreRestaurant.java
package esprit.DevUp.FoRest.Entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.*;

import javax.persistence.*;
import java.io.Serializable;
import java.util.List;

@Getter
@Setter
@Entity
@ToString
@AllArgsConstructor
@NoArgsConstructor
public class OffreRestaurant implements Serializable  {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id_offre_restaurant", nullable = false)
    private Integer idOffreRestaurant;
    private String nameOffre;
    private Integer nbrDays;
    private Boolean BREAKFAST;
    private Boolean LUNCH ;
    private Boolean DINNER;
    private Integer price;

    @OneToMany(mappedBy = "offreRestaurant")
    private List<accessRestaurant> accessRestaurants;

    @ManyToOne
    @JsonIgnore
    Restaurant restaurant;

}
